Main.hs

```diff
@@ -11,28 +11,32 @@ import qualified Data.List as L
 import Text.Printf (printf)
 
 multByIndex :: [Int] -> [Int]
-multByIndex = undefined
+multByIndex xs = zipWith (*) xs [0..]
 
 powerByIndex :: [Int] -> [Int]
-powerByIndex = undefined
+powerByIndex xs = zipWith (^) xs [0..]
+
 
 productOfDifference :: [Int] -> Int
-productOfDifference = undefined
+productOfDifference xs
+    | null xs = error "The list is empty"
+    | xs == [head xs] = error "The list is too short"
+    | otherwise = foldl (*) 1 (zipWith (-) xs (tail xs))
```

Comment:
This is a very contrived way to do something trivial. Just pattern match on the list instead of using guards.